notebook/notebookapp.py

```diff
@@ -155,7 +155,7 @@ def init_settings(self, ipython_app, kernel_manager, contents_manager,
             "template_path",
             ipython_app.template_file_path,
         )
-        if isinstance(_template_path, str):
+        if isinstance(_template_path, py3compat.string_types):
```

Comment:
Also because checking for just a list would be insufficient, as we would also need to check for tuples. In both cases, there are at least two types to check (except on Python 3, where string_types is `(str,)`.